pkg/build/admission/strategyrestrictions/admission.go

```diff
@@ -116,6 +149,9 @@ func (a *buildByStrategy) checkBuildAuthorization(build *buildapi.Build, attr ad
 }
 
 func (a *buildByStrategy) checkBuildConfigAuthorization(buildConfig *buildapi.BuildConfig, attr admission.Attributes) error {
+	if err := a.setImplicitBuildPermission(&buildConfig.Spec.Strategy, attr); err != nil {
```

Comment:
This should happen after the strategy check. If I'm not allowed to create S2I builds, then it doesn't matter what the implicitBuildPermission is.